### ISSUE TITLE
Disable Style/SymbolProc in db/migrate? 

### ISSUE THREAD

**Issue by danielmorrison** [testdouble/standard#31]
I'm not sure this should be fixed, but it came up today with a Rails migration. 

Given this migration: 
```ruby
create_table :customers do |t|
  t.timestamps
end
```

It wants to autofix it (Style/SymbolProc cop) to:

```ruby
create_table :customers, &:timestamps
```

I get it, but in a migration it seems weird to stray from the `t.` DSL. 

I think `timestamps` is the only method without arguments, so this should be very rare. I hate to disable a cop for a whole directory, seems like overkill. Figured I'd raise the issue though.

**Comment by searls** [comment]
Had this happen to me. Felt the same way. Have no good solutions either. Would definitely rather not having a catch-all ignore for a cop based on file name but I realize this is silly

**Comment by jcraigk** [comment]
Is the mechanism of providing different .rubocop.yml files in sub directories supported here?  I often find I want slightly different rules based on folders, such as migrations or specs.  I realize folder-based config doesn't solve the problem entirely but it might help.  To support that, multiple style variations could be defined, one each each idiom like Rails Migrations, etc.

**Comment by danielmorrison** [comment]
@jcraigk yes, I could add this to my `.standard.yml`:

```yaml
ignore:
  - 'db/migrate/*':
    - Style/SymbolProc
```

Currently, I'm planning to live with that one migration being "ugly" in the interest of standardizing.

**Comment by searls** [comment]
It'd be a great help if you could advise how to detect (reliably) we are in a Rails app (see #27). I'd be comfortable with a one-off change pertaining to db/migrate if it was in the context of a code branch that had already identified we're mucking with things Because Rails™.

**Comment by twe4ked** [comment]
I'd vote to disable `Style/SymbolProc` everywhere. There are lots of DSLs where it doesn't make sense to collapse the proc.

An example of a DSL I've been working on recently. I wouldn't want to format these differently simply because one doesn't take arguments.

```ruby
with_aggregate(FooAggregate, command.aggregate_id) do |foo|
  foo.some_action_that_takes_no_args
end

with_aggregate(FooAggregate, command.aggregate_id) do |foo|
  foo.some_action(arg1: command.arg1)
end
```

Personally I like to use the shorthand syntax when iterating over a list. In most other cases I leave the full version.

**Comment by searls** [comment]
I'm amenable to disabling this rule whole-cloth, but need to think more about it. Anyone else have an opinion?

**Comment by davidcelis** [comment]
I'd also agree with disabling this rule entirely. The short-hand block syntax is nice, but there are too many examples where it makes less sense than expanding the block. Too hard to enforce reliably, IMO

**Comment by noelrappin** [comment]
I think I'd be onboard with disabling it entirely.

**Comment by searls** [comment]
Agreed. Shipped in 0.0.15
